### ISSUE TITLE
how to configure wpantund ?

### ISSUE THREAD

**Issue by Yug11** [makerdiary/nrf52840-mdk#17]
I am implementing the openthread on makerdiary's nRF52840-MDK using a Linux machine. 
It goes fine up to flashing the board. After that, the step where we have to configure wpantund, it constantly shows an error and it remains uninitialized. Help me resolve this issue .

**Comment by caizelin** [comment]
> I am implementing the openthread on makerdiary's nRF52840-MDK using a Linux machine.
> It goes fine up to flashing the board. After that, the step where we have to configure wpantund, it constantly shows an error and it remains uninitialized. Help me resolve this issue .

@Yug11 You can follow this tutorial: https://wiki.makerdiary.com/nrf52840-mdk/openthread/#configure-wpantund

**Comment by Yug11** [comment]
@caizelin 
I have been following the same tutorial but it is not working as at this step the terminal keeps on and nothing happens
![Screenshot from 2019-03-25 12-16-05](https://user-images.githubusercontent.com/41464747/54900030-cd506f00-4ef7-11e9-99fb-f7557deb2dbc.png)

sometimes it shows unable to open the socket and sometimes descriptor in a bad state

**Comment by caizelin** [comment]
Make sure which serial port name you are using. It seems that `/dev/ttyACH0` does not exist.

**Comment by Yug11** [comment]
but it does exist
![Screenshot from 2019-03-25 12-37-17](https://user-images.githubusercontent.com/41464747/54900895-e6a6ea80-4efa-11e9-9b91-75463f372ac1.png)

**Comment by caizelin** [comment]
> but it does exist
> ![Screenshot from 2019-03-25 12-37-17](https://user-images.githubusercontent.com/41464747/54900895-e6a6ea80-4efa-11e9-9b91-75463f372ac1.png)

Yes. But I saw you were using `ttyACH0` not `ttyACM0`. Did you try `ttyACM0` or `ttyACM1`?

**Comment by Yug11** [comment]
ttyACM0

**Comment by caizelin** [comment]
> ttyACM0

Then check the UART configuration before you compile openthread. In `openthread/examples/platforms/nrf52840/platform-config.h` file:
```
/**
 * @def UART_HWFC
 *
 * UART Hardware Flow Control.
 *
 * @brief Possible values:
 *         \ref NRF_UART_HWFC_ENABLED - HW Flow control enabled.
 *         \ref NRF_UART_HWFC_DISABLED - HW Flow control disabled.
 *
 */
#ifndef UART_HWFC
#define UART_HWFC NRF_UART_HWFC_DISABLED
#endif

/**
 * @def UART_PIN_TX
 *
 * UART TX Pin.
 *
 */
#ifndef UART_PIN_TX
#define UART_PIN_TX 20
#endif

/**
 * @def UART_PIN_RX
 *
 * UART RX Pin.
 *
 */
#ifndef UART_PIN_RX
#define UART_PIN_RX 19
#endif
```

**Comment by Yug11** [comment]
![Screenshot from 2019-03-25 12-46-39](https://user-images.githubusercontent.com/41464747/54901334-28846080-4efc-11e9-9979-74f40f1ce136.png)

I changed it to ttyACM0 then this happens

**Comment by Yug11** [comment]
> > ttyACM0
> 
> Then check the UART configuration before you compile openthread. In `openthread/examples/platforms/nrf52840/platform-config.h` file:
> 
> ```
> /**
>  * @def UART_HWFC
>  *
>  * UART Hardware Flow Control.
>  *
>  * @brief Possible values:
>  *         \ref NRF_UART_HWFC_ENABLED - HW Flow control enabled.
>  *         \ref NRF_UART_HWFC_DISABLED - HW Flow control disabled.
>  *
>  */
> #ifndef UART_HWFC
> #define UART_HWFC NRF_UART_HWFC_DISABLED
> #endif
> 
> /**
>  * @def UART_PIN_TX
>  *
>  * UART TX Pin.
>  *
>  */
> #ifndef UART_PIN_TX
> #define UART_PIN_TX 20
> #endif
> 
> /**
>  * @def UART_PIN_RX
>  *
>  * UART RX Pin.
>  *
>  */
> #ifndef UART_PIN_RX
> #define UART_PIN_RX 19
> #endif
> ```

Yes I did open this file and copied this configuration at the end

**Comment by Yug11** [comment]
> > ttyACM0
> 
> Then check the UART configuration before you compile openthread. In `openthread/examples/platforms/nrf52840/platform-config.h` file:
> 
> ```
> /**
>  * @def UART_HWFC
>  *
>  * UART Hardware Flow Control.
>  *
>  * @brief Possible values:
>  *         \ref NRF_UART_HWFC_ENABLED - HW Flow control enabled.
>  *         \ref NRF_UART_HWFC_DISABLED - HW Flow control disabled.
>  *
>  */
> #ifndef UART_HWFC
> #define UART_HWFC NRF_UART_HWFC_DISABLED
> #endif
> 
> /**
>  * @def UART_PIN_TX
>  *
>  * UART TX Pin.
>  *
>  */
> #ifndef UART_PIN_TX
> #define UART_PIN_TX 20
> #endif
> 
> /**
>  * @def UART_PIN_RX
>  *
>  * UART RX Pin.
>  *
>  */
> #ifndef UART_PIN_RX
> #define UART_PIN_RX 19
> #endif
> ```

@caizelin 
I need to add these lines at the end of the file or change the whole file??

**Comment by caizelin** [comment]
@Yug11 Just define `UART_HWFC` to `NRF_UART_HWFC_DISABLED`, `UART_PIN_TX` to `20`, `UART_PIN_RX` to `19`

**Comment by caizelin** [comment]
BTW, you can also try the pre-built ncp firmware [ot-ncp-ftd.hex](https://github.com/makerdiary/nrf52840-mdk/tree/master/firmware/openthread).

**Comment by Yug11** [comment]
@caizelin I have tried both the methods  but none of it worked


![Screenshot from 2019-03-25 15-06-10](https://user-images.githubusercontent.com/41464747/54909432-3217c380-4f10-11e9-8bf2-54c88bd7b715.png)

**Comment by Yug11** [comment]
once it went up to this stage and lagged, after that, I tried it again but the same error occurred

![Screenshot from 2019-03-25 13-41-25](https://user-images.githubusercontent.com/41464747/54909778-f3ced400-4f10-11e9-89bc-cfc584151380.png)
